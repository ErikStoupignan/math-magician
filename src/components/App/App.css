@font-face {
  font-family: 'calculator';

  /* src: url('../src/assets/fonts/Calculator.ttf') format('truetype'); */
  src: url('../../assets/fonts/Calculator.ttf') format('truetype');
  font-weight: bold;
}

.App {
  width: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: relative;
  background: linear-gradient(45deg, rgb(2, 2, 124), rgb(87, 181, 224), rgb(175, 236, 236));
}

.welcome {
  margin: 30px;
  position: absolute;
  top: 0;
  font-family: calculator, Courier, monospace;
  font-size: 5vw;
  color: rgb(0, 0, 0);
  text-shadow: 0 1px rgb(0, 0, 0);
}
