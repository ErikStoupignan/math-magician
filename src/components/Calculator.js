/* eslint-disable react/prefer-stateless-function */
/* eslint-disable arrow-body-style */
import React from 'react';
import '../styles/Calculator.css';
// import calculate from '../logic/calculate';

export default class Calculator extends React.Component {
  render() {
    return (
      <div className="Container-calculator">
        <p className="screen">13091996</p>
        <Buttons />
      </div>
    );
  }
}

const Buttons = () => {
  return (
    <div className="Container-buttons">
      <button type="button" className="btn0">0</button>
      <button type="button" className="btn1">1</button>
      <button type="button" className="btn2">2</button>
      <button type="button" className="btn3">3</button>
      <button type="button" className="btn4">4</button>
      <button type="button" className="btn5">5</button>
      <button type="button" className="btn6">6</button>
      <button type="button" className="btn7">7</button>
      <button type="button" className="btn8">8</button>
      <button type="button" className="btn9">9</button>
      <button type="button" className="btn-add orange">+</button>
      <button type="button" className="btn-less orange">-</button>
      <button type="button" className="btn-division orange">/</button>
      <button type="button" className="btn-multiplication orange">*</button>
      <button type="button" className="btn-percentage">%</button>
      <button type="button" className="btn-moremin">+/-</button>
      <button type="button" className="btn.dot">.</button>
      <button type="button" className="btn-equal orange">=</button>
      <button type="button" className="btn-Ac">AC</button>
    </div>
  );
};
