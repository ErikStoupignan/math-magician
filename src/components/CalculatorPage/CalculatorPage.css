.CalculatorPage-Container {
  display: flex;
  flex-wrap: wrap;
  margin: 10px;
}

.Container-title {
  width: 49%;
  height: auto;
}

.CalculatorPage-title {
  text-align: left;
}

.CalculatorPage-AppContainer {
  display: flex;
  justify-content: center;
  width: 50%;
  height: auto;
  min-width: 250px;
  min-height: 250px;
}

.CalculatorPage-paragraph {
  font-weight: 500;
  text-align: center;
  width: 49%;
}
