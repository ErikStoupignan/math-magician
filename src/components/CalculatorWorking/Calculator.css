
@font-face {
  font-family: 'calculator';
  src: url('../../assets/fonts/Calculator.ttf') format('truetype');
  font-weight: bold;
}

* {
  margin: 0;
  padding: 0;
}

.Container-calculator {
  width: 30vw;
  height: 30vw;
  box-shadow: 0 0 100px black;
  border-radius: 30px 30px 30px 30px;
  margin-top: 30px;
}

.screen {
  height: 5vw;
  display: flex;
  justify-content: right;
  align-items: center;
  font-family: Calculator, Courier, monospace;
  font-size: 30px;
  background-color: rgb(221, 220, 220);
  padding: 0 40px 0 10px;
  border-radius: 30px 30px 0 0;
}

.Container-buttons {
  height: 25vw;
  display: grid;
  grid-template-areas:
    "ac masmin percentage division"
    "seven eight nine multiplication"
    "four five six less"
    "one two three add"
    "zero zero dot equal";
  background-color: #000;
  border-radius: 0 0 30px 30px;
}

.Container-buttons button {
  border: none;
  font-size: 20px;
  transition: 0.1s;
  cursor: pointer;
}

.Container-buttons button:hover {
  transform: scale(1.1);
  background-color: rgb(231, 226, 226);
}

.Container-buttons button:active {
  transform: scale(1.1);
  background-color: rgba(151, 150, 150, 0.61);
  color: aliceblue;
}

.btn0 {
  grid-area: zero;
  border-radius: 0 0 0 30px;
}
.btn1 { grid-area: one; }
.btn2 { grid-area: two; }
.btn3 { grid-area: three; }
.btn4 { grid-area: four; }
.btn5 { grid-area: five; }
.btn6 { grid-area: six; }
.btn7 { grid-area: seven; }
.btn8 { grid-area: eight; }
.btn9 { grid-area: nine; }
.btn-add { grid-area: add; }
.btn-less { grid-area: less; }
.btn-division { grid-area: division; }
.btn-multiplication { grid-area: multiplication; }
.btn-percentage { grid-area: percentage; }
.btn-moremin { grid-area: masmin; }
.btn.dot { grid-area: dot; }

.btn-equal {
  grid-area: equal;
  border-radius: 0 0 30px 0;
}
.btn-Ac { grid-area: ac; }
.orange { background-color: orange; }
.orange:hover { background-color: rgb(253, 192, 78) !important; }
