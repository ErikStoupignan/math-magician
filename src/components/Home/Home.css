.Home-Container {
  display: flex;
  flex-direction: column;
  margin: 10px;
}

.Home-title {
  text-align: left;
  width: fit-content;
  margin-bottom: 20px;
}

.Home-paragraph {
  display: flex;
  text-align: justify;
}
