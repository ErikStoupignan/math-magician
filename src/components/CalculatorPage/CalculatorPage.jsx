import React from 'react';
import Calculator from '../CalculatorWorking/Calculator';
import './CalculatorPage.css';

const CalculatorPage = () => (
  <section>
    <div>
      <h2>Let&apos;s do some Math!</h2>
    </div>
    <div>
      <Calculator />
    </div>
  </section>
);

export default CalculatorPage;
