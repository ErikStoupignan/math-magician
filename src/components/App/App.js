import { BrowserRouter, Routes, Route } from 'react-router-dom';
import NavBar from '../NavBar/NavBar';
import CalculatorPage from '../CalculatorPage/CalculatorPage';
import Home from '../Home/Home';
import Quote from '../Quote/Quote';
import './App.css';

const App = () => (
  <BrowserRouter>
    <NavBar />
    <Routes>
      <Route path="/" element={<Home />} />
      <Route path="/Home" element={<Home />} />
      <Route path="/CalculatorPage" element={<CalculatorPage />} />
      <Route path="/Quote" element={<Quote />} />
    </Routes>
  </BrowserRouter>
);

export default App;
