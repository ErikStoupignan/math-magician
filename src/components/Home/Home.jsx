import './Home.css';
import React from 'react';

const Home = () => (
  <section className="Home-Container">
    <div>
      <h2 className="Home-title">Welcome to our page</h2>
    </div>
    <p className="Home-paragraph">
      Placeat quisquam debitis consequuntur. Lorem ipsum dolor,
      sit amet consectetur adipisicing elit.
      Laborum ducimus culpa nemo facere sequi earum voluptatibus!
      Necessitatibus temporibus numquam dignissimos architecto odit sed optio quod tenetur,
      fugiat unde tempora. Cupiditate!
      <br />
      <br />
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
      Officia asperiores harum aspernatur magni earum ullam culpa,
      pariatur officiis reprehenderit minus sit,
      porro nesciunt! Quidem, cupiditate.
    </p>

  </section>
);

export default Home;
