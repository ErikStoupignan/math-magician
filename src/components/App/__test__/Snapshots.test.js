import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Home from '../../Home/Home';
import App from '../App';
import CalculatorPage from '../../CalculatorPage/CalculatorPage';
import Quote from '../../Quote/Quote';
import Calculator from '../../CalculatorWorking/Calculator';

// Snapshots test
test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText('Welcome to our page');
  expect(linkElement).toBeInTheDocument();
});

test('Snapshot testing on Home page', () => {
  const tree = render(<Home />);
  expect(tree).toMatchSnapshot();
});

test('Snapshot testing on CalculatorPage page', () => {
  const tree = render(<CalculatorPage />);
  expect(tree).toMatchSnapshot();
});

test('Snapshot testing on Quote page', () => {
  const tree = render(<Quote />);
  expect(tree).toMatchSnapshot();
});

// Simulate user interaction test
test('User event on Rack Dom', () => {
  render(<Calculator />);
  userEvent.click(screen.getByText('1'));
  userEvent.click(screen.getByText('2'));
  userEvent.click(screen.getByText('+'));
  userEvent.click(screen.getByText('3'));
  userEvent.click(screen.getByText('='));
  // 12 + 3 = 15  <- on display
  expect(screen.getByTestId('display')).toHaveTextContent('15');
});
