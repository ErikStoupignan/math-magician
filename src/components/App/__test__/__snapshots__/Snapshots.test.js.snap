// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing on CalculatorPage page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="CalculatorPage-Container"
      >
        <div
          class="Container-title"
        >
          <h2
            class="CalculatorPage-title"
          >
            Let's do some Math!
          </h2>
          <p
            class="CalculatorPage-paragraph"
          >
            Use your own keyboard!
          </p>
        </div>
        <div
          class="CalculatorPage-AppContainer"
        >
          <div
            class="Container-calculator"
          >
            <p
              class="screen"
              data-testid="display"
            >
              Welcome
            </p>
            <div
              class="Container-buttons"
            >
              <button
                class="btn0"
                type="button"
              >
                0
              </button>
              <button
                class="btn2"
                type="button"
              >
                2
              </button>
              <button
                class="btn1"
                type="button"
              >
                1
              </button>
              <button
                class="btn3"
                type="button"
              >
                3
              </button>
              <button
                class="btn4"
                type="button"
              >
                4
              </button>
              <button
                class="btn5"
                type="button"
              >
                5
              </button>
              <button
                class="btn6"
                type="button"
              >
                6
              </button>
              <button
                class="btn7"
                type="button"
              >
                7
              </button>
              <button
                class="btn8"
                type="button"
              >
                8
              </button>
              <button
                class="btn9"
                type="button"
              >
                9
              </button>
              <button
                class="btn-add orange"
                type="button"
              >
                +
              </button>
              <button
                class="btn-less orange"
                type="button"
              >
                -
              </button>
              <button
                class="btn-division orange"
                type="button"
              >
                รท
              </button>
              <button
                class="btn-multiplication orange"
                type="button"
              >
                *
              </button>
              <button
                class="btn-percentage"
                type="button"
              >
                %
              </button>
              <button
                class="btn-moremin"
                type="button"
              >
                +/-
              </button>
              <button
                class="btn.dot"
                type="button"
              >
                .
              </button>
              <button
                class="btn-equal orange"
                type="button"
              >
                =
              </button>
              <button
                class="btn-Ac"
                type="button"
              >
                AC
              </button>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="CalculatorPage-Container"
    >
      <div
        class="Container-title"
      >
        <h2
          class="CalculatorPage-title"
        >
          Let's do some Math!
        </h2>
        <p
          class="CalculatorPage-paragraph"
        >
          Use your own keyboard!
        </p>
      </div>
      <div
        class="CalculatorPage-AppContainer"
      >
        <div
          class="Container-calculator"
        >
          <p
            class="screen"
            data-testid="display"
          >
            Welcome
          </p>
          <div
            class="Container-buttons"
          >
            <button
              class="btn0"
              type="button"
            >
              0
            </button>
            <button
              class="btn2"
              type="button"
            >
              2
            </button>
            <button
              class="btn1"
              type="button"
            >
              1
            </button>
            <button
              class="btn3"
              type="button"
            >
              3
            </button>
            <button
              class="btn4"
              type="button"
            >
              4
            </button>
            <button
              class="btn5"
              type="button"
            >
              5
            </button>
            <button
              class="btn6"
              type="button"
            >
              6
            </button>
            <button
              class="btn7"
              type="button"
            >
              7
            </button>
            <button
              class="btn8"
              type="button"
            >
              8
            </button>
            <button
              class="btn9"
              type="button"
            >
              9
            </button>
            <button
              class="btn-add orange"
              type="button"
            >
              +
            </button>
            <button
              class="btn-less orange"
              type="button"
            >
              -
            </button>
            <button
              class="btn-division orange"
              type="button"
            >
              รท
            </button>
            <button
              class="btn-multiplication orange"
              type="button"
            >
              *
            </button>
            <button
              class="btn-percentage"
              type="button"
            >
              %
            </button>
            <button
              class="btn-moremin"
              type="button"
            >
              +/-
            </button>
            <button
              class="btn.dot"
              type="button"
            >
              .
            </button>
            <button
              class="btn-equal orange"
              type="button"
            >
              =
            </button>
            <button
              class="btn-Ac"
              type="button"
            >
              AC
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot testing on Home page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="Home-Container"
      >
        <div>
          <h2
            class="Home-title"
          >
            Welcome to our page
          </h2>
        </div>
        <p
          class="Home-paragraph"
        >
          Placeat quisquam debitis consequuntur. Lorem ipsum dolor, sit amet consectetur adipisicing elit. Laborum ducimus culpa nemo facere sequi earum voluptatibus! Necessitatibus temporibus numquam dignissimos architecto odit sed optio quod tenetur, fugiat unde tempora. Cupiditate!
          <br />
          <br />
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Officia asperiores harum aspernatur magni earum ullam culpa, pariatur officiis reprehenderit minus sit, porro nesciunt! Quidem, cupiditate.
        </p>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="Home-Container"
    >
      <div>
        <h2
          class="Home-title"
        >
          Welcome to our page
        </h2>
      </div>
      <p
        class="Home-paragraph"
      >
        Placeat quisquam debitis consequuntur. Lorem ipsum dolor, sit amet consectetur adipisicing elit. Laborum ducimus culpa nemo facere sequi earum voluptatibus! Necessitatibus temporibus numquam dignissimos architecto odit sed optio quod tenetur, fugiat unde tempora. Cupiditate!
        <br />
        <br />
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Officia asperiores harum aspernatur magni earum ullam culpa, pariatur officiis reprehenderit minus sit, porro nesciunt! Quidem, cupiditate.
      </p>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot testing on Quote page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="Quote-Container"
      >
        <p
          class="Quote-paragraph"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum dolor sit amet consectetur adipisicing elit. Similique, voluptates distinctio veritatis aperiam rem, laboriosam voluptate error vero consequuntur nostrum nam, expedita sed doloremque voluptatum. Voluptatum molestiae distinctio modi! Deserunt! - Lorem
        </p>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="Quote-Container"
    >
      <p
        class="Quote-paragraph"
      >
        Lorem ipsum, dolor sit amet consectetur adipisicing elit. Lorem ipsum dolor sit amet consectetur adipisicing elit. Similique, voluptates distinctio veritatis aperiam rem, laboriosam voluptate error vero consequuntur nostrum nam, expedita sed doloremque voluptatum. Voluptatum molestiae distinctio modi! Deserunt! - Lorem
      </p>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
